### DISCUSSION TITLE
New UX + Performance improvements

### DISCUSSION THREAD

**Discussion by assafelovic** [assafelovic/gpt-researcher#974] [Announcements]
Excited to introduce a complete revamp to our React application which now provides a much smoother and optimal research experience than before. In addition we have some awesome new features like chatting with your reports, improved pubmed retrieval, additional embedding providers and more! As always, this is many thanks to our incredible community. Keep it coming! 

https://github.com/user-attachments/assets/78707236-120f-4a50-a4f7-6a225b98933a

## What's Changed
* NextJS Upgrades by @ElishaKay in https://github.com/assafelovic/gpt-researcher/pull/951
* Updated Contributing.md by @HimangshuYadav in https://github.com/assafelovic/gpt-researcher/pull/952
* Documentation update to correct LLM provider prefix for Azure OpenAI. by @scchengaiah in https://github.com/assafelovic/gpt-researcher/pull/958
* Updates code of conduct and removes all the grammatical errors. by @HimangshuYadav in https://github.com/assafelovic/gpt-researcher/pull/956
* Updates Readme.md (English version) by @HimangshuYadav in https://github.com/assafelovic/gpt-researcher/pull/955
* Sort pubmed_central search results by relevance versus default newest by @cannin in https://github.com/assafelovic/gpt-researcher/pull/953
* Add function get_similar_content_by_query_with_vectorstore.  by @dzerkes in https://github.com/assafelovic/gpt-researcher/pull/961
* Searx retriever refactoring by @igochkov in https://github.com/assafelovic/gpt-researcher/pull/960
* Update gptr-logs-handler.py by @DhruvKadam-git in https://github.com/assafelovic/gpt-researcher/pull/962
* Chat with History by @khoangothe in https://github.com/assafelovic/gpt-researcher/pull/885
* nextjs chat fix by @ElishaKay in https://github.com/assafelovic/gpt-researcher/pull/972
* Add embedding providers by @kesamet in https://github.com/assafelovic/gpt-researcher/pull/965
* update `getHost` reference by @emmanuel-ferdman in https://github.com/assafelovic/gpt-researcher/pull/973
* Improved NextJS UX by @assafelovic in https://github.com/assafelovic/gpt-researcher/pull/968

## New Contributors
* @HimangshuYadav made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/952
* @scchengaiah made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/958
* @cannin made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/953
* @dzerkes made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/961
* @igochkov made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/960
* @DhruvKadam-git made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/962
* @emmanuel-ferdman made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/973

**Full Changelog**: https://github.com/assafelovic/gpt-researcher/compare/v3.1.2...v3.1.3

<hr /><em>This discussion was created from the release <a href='https://github.com/assafelovic/gpt-researcher/releases/tag/v3.1.3'>New UX + Performance improvements</a>.</em>

**Comment by coreylansdell** [comment]
I've tried installing locally. I've tried through homebrew. I've tried following the directions here. I've gotten it on my browser but I have been unable to get it to do the work. 

One install allowed me to start the research but kept closing down because it was using too many tokens. I tried to modify the yml files and py files that it would only run 8000 tokens and throttle it... didn't work. 

That was the closes I got to it working. The rest of the installs got me to the front end and the back end did nothing. 

I am on a macbook pro M1 running sonoma 14.5 

Any tips at all would be sooo appreciated! I'm at a loss.

**Reply by coreylansdell** [reply]
I mean it looks really really useful. I wish I could run it.
